Sources/Context/ContextValueContainerKey.swift

```diff
@@ -34,7 +34,14 @@
 ///             }
 ///         }
 ///     }
-public protocol BaggageKey {
+///
+/// It is also generally considered appropriate to define a new protocol that conforms to `Context` that provides access to the property rather than forcing
+/// access to underlying storage.
+///
+///     protocol LoggerContextCarrier: Context {
+///         var logger: Logger { get }
+///     }
```

Comment:
This is the carriers proposal which is status quo (current state of the main branch) and we've found it to break down in real usage patterns.